### PR TITLE
K8s labels annotations

### LINKED ISSUE #62: Add labels to target Secrets created by secrets-manager operator
We have been using secrets-manager operator for a while to sync our `Secrets` from Vault with no issues (gitops philosohy).

Now we are using prometheus-operator with an additional scraping config in a `Secret` object, and we wanted to use secrets-manager operator to manage it (like a normal `Secret`).

The problem we have right now is that we need to add a specific label to the final `Secret` that prometheus-operator instance uses to scrape resources (in that case scrape `Secrets` with additional scraper configs).

We have checked the docs/code of secrets-manager operator, and we haven't seen how to set labels for target `Secrets` created by secrets-manager (a part from `lastUpdatedAt` and `managedBy` labels).

We wonder if there is any way to:
- Add labels for target secrets on `SecretDefinition` spec?
- Or maybe use `SecretDefinition` possible labels (if they exist) to add them to target `Secrets` ?

Thanks in advance!

### LINKED ISSUE #71: Upgrade apiVersion of CRD to v1 instead of v1beta1
CRDs have been graduated from v1beta1 to v1 in the kubernetes API but not in your CRD.

### PR DESCRIPTION
**Pull request by eduardogr**
# Status

READY

# Migrations

Yes,
Migrating from **apiextensions.k8s.io/v1beta1** to **apiextensions.k8s.io/v1**

# Description

- Resolving conflicts on PR #64 
  -  Populating annotations and labels from secret definition to the created secret 
  - Thanks to @stevendborrelli
- Upgrading apiextensions.k8s.io/v1beta1 to apiextensions.k8s.io/v1

# List of fixes # (issue)
  - fix #71 
  - fix #62 

# Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?
with secrets-manager tests

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/v1alpha1/groupversion_info.go [hunk 1]
```diff
@@ -23,9 +23,14 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/scheme"
```
Comment (eduardogr) [inline api/v1alpha1/groupversion_info.go:23]: changes from #64

File: controllers/secretdefinition_controller.go [hunk 2]
```diff
@@ -35,7 +35,9 @@ import (
 const (
 	// https://golang.org/pkg/time/#pkg-constants
 	timestampFormat = "2006-01-02T15.04.05Z"
-	finalizerName   = "secret.finalizer.secrets-manager.tuenti.io"
+	finalizerName   = "secret.finalizer." + smv1alpha1.Group
+	managedByLabel  = "app.kubernetes.io/managed-by"
+	lastUpdateLabel = smv1alpha1.Group + "/lastUpdateTime"
```
Comment (eduardogr) [inline controllers/secretdefinition_controller.go:40]: changes from #64

File: controllers/secretdefinition_controller_test.go [hunk 3]
```diff
@@ -12,8 +12,10 @@ import (
```
Comment (eduardogr) [inline controllers/secretdefinition_controller_test.go:12]: I'll will take more care of the test suite in the next steps. https://github.com/tuenti/secrets-manager/issues/79

**Comment by eduardogr** [comment]
@stevendborrelli i have here your original commits.

After your commits i've added some refactor and a fix for the test you've added.
Are you Ok with this?

Thanks a lot for your changes.

**Comment by stevendborrelli** [comment]
Looks good to me! Thanks for fixing the merge conflict.

**Comment by eduardogr** [comment]
Thanks a lot @stevendborrelli  :)

**Review by baudlord — APPROVED** [review]

**Review by kamunas — APPROVED** [review]

**Comment by raelga** [comment]
Thanks for implementing this!

We have been testing it and we faced a couple of issues. The first one is the usage of the `v1` without updating the schema nor the controller toolchain (discussed [here](https://github.com/tuenti/secrets-manager/issues/71#issuecomment-801405025).

Regarding this feature implementation using metadata propagation, for our particular use case, it leads to an undesired behaviour that we didn't expected.

In some scenarios, labels are used to _determine ownership_ of resources, especially when they are managed from `outside` the cluster. For example, some `GitOps` operators, like ArgoCD, use labels to identify what resources are part of a particular ArgoCD application. Propagating the labels to the resources created by the `SecretDefinition` instance will _transfer the ownership_ of the `Secrets` created to ArgoCD.

When the `SecretDefinition`, created by the GitOps operator, propagates the labels, it adds metadata to the `Secret` resource that will end up with ArgoCD taking the "ownership" of the resource. In a GitOps scenario, that leads to the deletion of the resources, as is not present in the repository. That triggers the recreation of the secret by the `secrets-manager` operator, and the loop goes on. Using proper `ownerReferences` is not viable as in multi-cluster scenarios, the GitOps operator manage the resources from an external cluster. This is one of the scenarios that we found, put probably there are other scenarios where this propagation may lead to an undesired behaviour.

**Pull request merged**